Document/0x06i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -347,6 +347,97 @@ func dosomething(argumentx:TypeX) throws {
 
 - Use the `try!` expression to assert that the error won't occur.
 
+
+- Handle the generic error as a `Result` return:
+
+```swift
+enum ErrorType: Error {
+    case typeOne
+    case typeTwo
+}
+
+func functionWithResult(param: String?) -> Result<String, ErrorType> {
+    guard let value = param else {
+        return .failure(.typeOne)
+    }
+    return .success(value)
+}
+
+func callResultFunction() {
+    let result = functionWithResult(param: "OWASP")
+
+    switch result {
+    case let .success(value):
+        print(value)
+    case let .failure(error):
+        print("Error: \(error)")
+    }
+}
+```
+
+- Handle network and JSON decoding errors with a `Result` type:
+
+```swift 
+struct MSTG: Codable {
+    var root: String
+    var plugins: [String]
+    var structure: MSTGStructure
+    var title: String
+    var language: String
+    var description: String
+}
+
+struct MSTGStructure: Codable {
+    var readme: String
+}
+
+enum RequestError: Error {
+    case requestError(Error)
+    case noData
+    case jsonError
+}
+
+func getMSTGInfo() {
+    guard let url = URL(string: "https://raw.githubusercontent.com/OWASP/owasp-mstg/master/book.json") else {
+        return
+    }
+
+    request(url: url) { result in
+        switch result {
+        case let .success(data):
+            print("OWASP MSTG title: \(data.title)")
+            print("OWASP MSTG description: \(data.description)")
+        case let .failure(error):
+            switch error {
+            case let .requestError(error):
+                print(error.localizedDescription)
+            case .noData:
+                print("No data received.")
+            case .jsonError:
+                print("Error parsing JSON")
+            }
```

Comment:
can you, instead of pirnting it, only provide the error as a resolution? let's try to be strict in that (in terms of exception handling)

Reply:
Rephrase: can you both vote in Slack to see if we want to keep the printing in for clarification? As long as we are consistent after that ^^. Great work @cak !

Reply:
Great idea, removed all `print` statements from examples.